tests/testthat/test-cube-transforms.R

```diff
@@ -884,26 +884,27 @@ test_that("subtotals after cube subsetting", {
 
     one_col <- cubify(
         c(
-            5, 0, 5,
-            12, 0, 12,
-            17, 0, 17,
-            7, 0, 7,
-            10, 0, 10,
-            12, 0, 12,
-            22, 0, 22
+            5,  5,  0,
+            12, 12, 0,
+            17, 17, 0,
+            7,  7,  0,
+            10, 10, 0,
+            12, 12, 0,
+            22, 22, 0
         ),
         dims = list(
             "feelings" =
                 c(
                     "extremely happy", "somewhat happy", "happy", "neutral",
                     "somewhat unhappy", "extremely unhappy", "unhappy"
                 ),
-            "animals" = c("dogs", "felines", "both")
+            "animals" = c("dogs", "both", "felines")
```

Comment:
This change looks good: before we must not have been dropping the category appropriately so the `felines` insertion was being collated ahead of the `both` insertion, even though by the rules of ordering all anchored-to-non-existant insertions should be at the bottom.

```
$animals
An object of class  “Transforms” 
$insertions
  anchor    name     func    args
1      1 felines subtotal       1
2      2    both subtotal 1 and 2
```

and:
```
> categories(variables(pet_feeling_both)[[2]])
  id    name value missing
1  1    cats     1   FALSE
2  2    dogs     2   FALSE
3 -1 No Data    NA    TRUE
```

Reply:
Yeah I think it was caused by an error of mine in using the `nargs() = 2` to identify cases where only `i` was supplied as an index and not `j` or `...`, this worked in most cases but of course if you provided `cube[1, drop = FALSE]` you were sending three arguments, but only one index.